sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PushVariantIntoScan.scala

```diff
@@ -244,6 +272,24 @@ class VariantInRelation {
         // Rewrite the attribute in advance, rather than depending on the last branch to rewrite it.
         // Ww need to avoid the `v@StructPathToVariant(fields)` branch to rewrite the child again.
         GetStructField(rewriteAttribute(v), fields(RequestedVariantField(g)))
+      case s @ StaticInvoke(
```

Comment:
This is likely the most annoying part of `RuntimeReplaceable`: pattern match becomes less intuitive.

Now I think it's probably better not to use `RuntimeReplaceable` for expressions that need be matched post analysis, until we improve `RuntimeReplaceable` to be replaced truly at runtime.

Reply:
Let's hold it first, and I'll try to see how to make a truly `RuntimeReplaceable` (as we discussed privately, I think its idea is to only execute the `replaced expression` at `runtime`, while still maintaining it during the analysis and optimization stages, right?)

Reply:
Yes, that will be the best.